src/ada/src/services/arv/automation_request_validator.ads

```diff
@@ -0,0 +1,277 @@
+with Ada.Containers.Formal_Doubly_Linked_Lists;
+with Common; use Common;
+with Ada.Containers.Functional_Maps;
+with Ada.Containers.Formal_Hashed_Maps;
+with Ada.Containers; use all type Ada.Containers.Count_Type;
+with LMCP_Messages; use LMCP_Messages;
+with Automation_Request_Validator_Communication; use Automation_Request_Validator_Communication;
+
+--  Package containing the fonctionality of the service. It uses its communication
+--  counter-part to send and receive messages.
+
+package Automation_Request_Validator with SPARK_Mode is
+   pragma Unevaluated_Use_Of_Old (Allow);
+
+   --  Configuration data is separated from the service state as it is not
+   --  handled by the same primitives. We use functional containers, as it is
+   --  not supposed to be modified often.
+
+   type OperatingRegionAreas is record
+      KeepInAreas  : Int64_Seq;
+      KeepOutAreas : Int64_Seq;
+   end record;
+
+   package Int64_Operating_Region_Maps is new Ada.Containers.Functional_Maps
+     (Key_Type     => Int64,
+      Element_Type => OperatingRegionAreas);
+   type Operating_Region_Map is new Int64_Operating_Region_Maps.Map;
+
+   type Task_Kind is (AngledAreaSearchTask, ImpactLineSearchTask, ImpactPointSearchTask, Other_Task);
```

Comment:
Let's use underscores and Ada style casing, throughout. Thus Angled_Area_Search_Task etc.